client/me/help/help-contact-form/index.jsx

```diff
@@ -61,6 +70,22 @@ module.exports = React.createClass( {
 		};
 	},
 
+	componentWillReceiveProps: function( nextProps ) {
+		if ( isEqual( nextProps.valueLink.value, this.state ) ) {
+			return;
+		}
+
+		this.setState( nextProps.valueLink.value );
+	},
+
+	componentDidUpdate: function( prevProps, prevState ) {
+		if ( isEqual( prevState, this.state ) ) {
+			return;
+		}
+
+		this.props.valueLink.requestChange( Object.assign( {}, this.state ) );
```

Comment:
@omarjackman that's a good safety, but in my opinion a bit pre-emptive. do we know it's getting modified somewhere? if it is, we should focus on fixing that function because otherwise we will run ourselves crazy trying to protect against unknowns